bulla-contracts/src/functions/common.ts

```diff
@@ -160,3 +160,31 @@ export const getLatestPrice = (event: ethereum.Event): BigInt => {
   const bullaFactoringContract = BullaFactoring.bind(event.address);
   return bullaFactoringContract.pricePerShare();
 };
+
+export const getOrCreateHistoricalFactoringStatistics = (event: ethereum.Event): HistoricalFactoringStatistics => {
+  let historicalFactoringStatistics = HistoricalFactoringStatistics.load(event.address.toHexString());
+  const bullaFactoringContract = BullaFactoring.bind(event.address);
+  const fundInfo = bullaFactoringContract.getFundInfo();
+
+  if (!historicalFactoringStatistics) {
+    historicalFactoringStatistics = new HistoricalFactoringStatistics(event.address.toHexString());
+    historicalFactoringStatistics.address = event.address;
+    historicalFactoringStatistics.statistics = [];
+  }
+
+  const historyEntryId = historicalFactoringStatistics.id.concat("-").concat(event.block.timestamp.toString());
```

Comment:
I'm not sure concatenating the timestamp is good enough for this, ideal would be pool address + txHash + log index

Reply:
actually its fine, since ath same txHash, they will have the same pool values.